### PR TITLE
Modify Device.udiCarrier

### PR DESCRIPTION
**Pull request by bcastillomartel-va**

### CODE REVIEW

File: us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/Device.java [hunk 1]
```diff
@@ -129,6 +134,17 @@ public class Device implements AsList<Device>, DomainResource, HasTargetPatient
 
   @Valid Reference parent;
 
+  @JsonIgnore
+  @SuppressWarnings("unused")
+  @AssertTrue(message = "udiCarrier must contain at most one item.")
+  private boolean isValidUdiCarrier() {
+    if (udiCarrier.size() > 1) {
+      return false;
```
Comment (jblefkove-va) [inline us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/Device.java:142]: simplify to `return <statement>`

File: us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/Device.java [hunk 2]
```diff
@@ -129,6 +134,17 @@ public class Device implements AsList<Device>, DomainResource, HasTargetPatient
 
   @Valid Reference parent;
 
+  @JsonIgnore
+  @SuppressWarnings("unused")
+  @AssertTrue(message = "udiCarrier must contain at most one item.")
+  private boolean isValidUdiCarrier() {
+    if (udiCarrier.size() > 1) {
```
Comment (jblefkove-va) [inline us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/Device.java:141]: `udiCarrier` may be null

Comment (bcastillomartel-va) [reply us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/Device.java:141]: should a null `udiCarrier` trigger an error?

Comment (jblefkove-va) [reply us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/Device.java]: No, null/empty `udiCarrier` is valid

File: us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/Device.java [hunk 3]
```diff
@@ -129,6 +131,16 @@ public class Device implements AsList<Device>, DomainResource, HasTargetPatient
...
   @Valid Reference parent;
 
+  /* In R4 Device https://hl7.org/fhir/R4/device.html udiCarrier is modeled as a list, while in
+   * US Core R4 Implantable Device, it is modeled as a singular element.
+   * uidCarrier is modeled to use a list that contains at most one item. */
+  @JsonIgnore
+  @SuppressWarnings("unused")
+  @AssertTrue(message = "udiCarrier must contain at most one item.")
+  private boolean isValidUdiCarrier() {
+    return !(udiCarrier.size() > 1);
```
Comment (jblefkove-va) [inline us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/Device.java:141]: `!statement` is more difficult to understand, can we distribute the `!`?

File: us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/Device.java [hunk 4]
```diff
@@ -129,6 +132,16 @@ public class Device implements AsList<Device>, DomainResource, HasTargetPatient
...
   @Valid Reference parent;
 
+  /* In R4 Device https://hl7.org/fhir/R4/device.html udiCarrier is modeled as a list, while in
+   * US Core R4 Implantable Device, it is modeled as a singular element.
+   * uidCarrier is modeled to use a list that contains at most one item. */
+  @JsonIgnore
+  @SuppressWarnings("unused")
+  @AssertTrue(message = "udiCarrier must contain at most one item.")
+  private boolean isValidUdiCarrier() {
+    return Safe.list(udiCarrier).size() == 1;
```
Comment (jblefkove-va) [inline us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/Device.java:142]: It's valid for the list to be empty

**Review by jblefkove-va — APPROVED** [review]

**Pull request merged**